Type definitions break in TypeScript 2.7

The type definitions fail on latest version of TypeScript (tested with 2.7.2).

```
node_modules/restyped-axios/dist/index.d.ts
(3,157): Type 'API[Path][Method]' does not satisfy the constraint 'RestypedRoute'.
```

To recreate:

```
$ cat x.ts
import axios from 'restyped-axios'
$ tsc --version
Version 2.7.2
$ cat tsconfig.json
{
  "compilerOptions": {
    "target": "es6",
    "module": "commonjs",
    "noImplicitAny": false,
    "lib": [
      "es5",
      "es2015",
      "es6",
      "dom",
      "es2017.object"
    ],
    "typeRoots": [
      "node_modules/@types"
    ],
    "sourceMap": true,
    "jsx": "react",
    "allowJs": true,
    "experimentalDecorators": true,
    "forceConsistentCasingInFileNames": true,
    "emitDecoratorMetadata": true,
    "outDir": "./dist/",
    "baseUrl": ".",
    "paths": {
      "@api.d/*": ["../api.d/*"],
      "@api/*": ["./src/*"]
    }
  },
  "include": [
    "./src/**/*"
  ],
  "typeRoots": [
    "node_modules/@types"
  ]
}
$ tsc x.ts 
node_modules/restyped-axios/dist/index.d.ts(3,157): error TS2344: Type 'API[Path][Method]' does not satisfy the constraint 'RestypedRoute'.
node_modules/restyped-axios/dist/index.d.ts(9,152): error TS2344: Type 'API[Path][Method]' does not satisfy the constraint 'RestypedRoute'.
```

--- a/index.ts
+++ b/index.ts
@@ -6,10 +6,10 @@ import axios, {
   CancelTokenStatic
 } from 'axios'
 
-import {RestypedBase, RestypedRoute} from 'restyped'
+import {RestypedBase, RestypedIndexedBase, RestypedRoute} from 'restyped'
 
 export interface TypedAxiosRequestConfig<
-  API extends RestypedBase,
+  API extends RestypedIndexedBase,
   Path extends keyof API,
   Method extends keyof API[Path],
   RouteDef extends RestypedRoute = API[Path][Method]
@@ -21,7 +21,7 @@ export interface TypedAxiosRequestConfig<
 }
 
 export interface TypedAxiosResponse<
-  API extends RestypedBase,
+  API extends RestypedIndexedBase,
   Path extends keyof API,
   Method extends keyof API[Path],
   RouteDef extends RestypedRoute = API[Path][Method]

